integration-tests/relayinterface/chain_components_test.go

```diff
@@ -100,171 +118,350 @@ func DisableTests(it *SolanaChainComponentsInterfaceTester[*testing.T]) {
+					// setup call data
+					batchGetLatestValueRequest[bound1] = append(batchGetLatestValueRequest[bound1], types.BatchRead{ReadName: MethodTakingLatestParamsReturningTestStruct, Params: &LatestParams{I: 1 + i}, ReturnVal: &TestStruct{}})
+					batchGetLatestValueRequest[bound2] = append(batchGetLatestValueRequest[bound2], types.BatchRead{ReadName: MethodTakingLatestParamsReturningTestStruct, Params: &LatestParams{I: 1 + i}, ReturnVal: &TestStruct{}})
+				}
+				batchContractWrite(t, it, cw, bindings, batchCallEntry)
+
+				ctx := tests.Context(t)
+				require.NoError(t, cr.Bind(ctx, bindings))
+
+				result, err := cr.BatchGetLatestValues(ctx, batchGetLatestValueRequest)
+				require.NoError(t, err)
+
+				for idx := 0; idx < 10; idx++ {
+					fmt.Printf("expected: %+v\n", batchCallEntry[bound1][idx].ReturnValue)
+					if val, err := result[bound1][idx].GetResult(); err == nil {
+						fmt.Printf("result: %+v\n", val)
+					}
+				}
+
+				for i := 0; i < 10; i++ {
+					testDataAnyContract, testDataAnySecondContract := batchCallEntry[bound1], batchCallEntry[bound2]
+					resultAnyContract, resultAnySecondContract := result[bound1], result[bound2]
+					returnValueAnyContract, errAnyContract := resultAnyContract[i].GetResult()
+					returnValueAnySecondContract, errAnySecondContract := resultAnySecondContract[i].GetResult()
+					assert.NoError(t, errAnyContract)
+					assert.NoError(t, errAnySecondContract)
+					assert.Contains(t, resultAnyContract[i].ReadName, MethodTakingLatestParamsReturningTestStruct)
+					assert.Contains(t, resultAnySecondContract[i].ReadName, MethodTakingLatestParamsReturningTestStruct)
+					assert.Equal(t, testDataAnyContract[i].ReturnValue, returnValueAnyContract)
+					assert.Equal(t, testDataAnySecondContract[i].ReturnValue, returnValueAnySecondContract)
+				}
+			},
+		},
+	}
 
 	RunTests(t, it, testCases)
 }
 
-func RunContractReaderInLoopTests[T TestingT[T]](t T, it ChainComponentsInterfaceTester[T]) {
+func RunContractReaderInLoopTests[T WrappedTestingT[T]](t T, it ChainComponentsInterfaceTester[T]) {
 	RunContractReaderInterfaceTests(t, it, false, true)
 
 	var testCases []Testcase[T]
 
 	RunTests(t, it, testCases)
 }
 
-type SolanaChainComponentsInterfaceTesterHelper[T TestingT[T]] interface {
+type SolanaChainComponentsInterfaceTesterHelper[T WrappedTestingT[T]] interface {
 	Init(t T)
 	RPCClient() *chainreader.RPCClientWrapper
 	Context(t T) context.Context
 	Logger(t T) logger.Logger
-	GetJSONEncodedIDL(t T) []byte
-	CreateAccount(t T, it SolanaChainComponentsInterfaceTester[T], value uint64) solana.PublicKey
+	GetPrimaryIDL(t T) []byte
+	GetSecondaryIDL(t T) []byte
+	CreateAccount(t T, it SolanaChainComponentsInterfaceTester[T], contractName string, value uint64, testStruct TestStruct) solana.PublicKey
 	TXM() *txm.TxManager
 	SolanaClient() *client.Client
 }
 
-type SolanaChainComponentsInterfaceTester[T TestingT[T]] struct {
+type WrappedTestingT[T any] interface {
+	TestingT[T]
+	Name() string
+}
```

Comment:
This is probably something we could add to `chainlink-common` but added here to avoid changes in another repo.